eng/devices/devices-shared.cake

```diff
@@ -8,3 +8,15 @@ if (string.Equals(TARGET, "uitest", StringComparison.OrdinalIgnoreCase))
     DEFAULT_PROJECT = "../../src/Controls/tests/UITests/Controls.AppiumTests.csproj";
     DEFAULT_APP_PROJECT = "../../src/Controls/samples/Controls.Sample.UITests/Controls.Sample.UITests.csproj";
 }
+
+void FailRunOnOnlyInconclusiveTests(string testResultsFile)
+{
+    // When all tests are inconclusive the run does not fail, check if this is the case and fail the pipeline so we get notified
+	var totalTestCount = XmlPeek(testResultsFile, "/test-run/@total");
+	var inconclusiveTestCount = XmlPeek(testResultsFile, "/test-run/@inconclusive");
+	
+	if (totalTestCount.Equals(inconclusiveTestCount))
+    {
+		throw new Exception("All tests are marked inconclusive, no tests ran. There is probably something wrong with running the tests.");
+	}
```

Comment:
We had a case where we updated the UITest dependency and all iOS tests came up as inconclusive. For NUnit this still passes and so we thought: hooray, all green!

To make sure we notice something might be up, check the results and fail the test run if this scenario comes up.

Reply:
Moved this into a separate PR #17971